integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java

```diff
@@ -221,8 +223,40 @@ void testReplicasTooHigh() {
 
       //check the desired completed, available and failed statuses
       checkStatus(domainName, "False", "False", "True");
-      testPassed = true;
+      String patchStr = "[{\"op\": \"replace\", "
+          + "\"path\": \"/spec/webLogicCredentialsSecret/name\", \"value\": \"weblogic-credentials-foo\"}]";
+      logger.info("PatchStr for domainHome: {0}", patchStr);
 
+      V1Patch patch = new V1Patch(patchStr);
+      assertTrue(patchDomainCustomResource(domainName, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+          "patchDomainCustomResource failed");
+      testUntil(
+          domainStatusReasonMatches(domainName, domainNamespace, "DomainInvalid"),
+          getLogger(),
+          "waiting for domain status condition reason DomainInvalid exists"
+      );
+      testUntil(
+          domainStatusReasonMatches(domainName, domainNamespace, "ReplicasTooHigh"),
+          getLogger(),
+          "waiting for domain status condition reason ReplicasTooHigh exists"
+      );
+      patchStr
+          = "["
+          + "{\"op\": \"replace\", \"path\": \"/spec/clusters/0/replicas\", \"value\": 2}"
+          + "]";
+      assertTrue(patchDomainCustomResource(domainName, domainNamespace, patch, V1Patch.PATCH_FORMAT_JSON_PATCH),
+          "patchDomainCustomResource failed");
+      testUntil(
+          domainStatusReasonMatches(domainName, domainNamespace, "DomainInvalid"),
+          getLogger(),
+          "waiting for domain status condition reason DomainInvalid exists"
+      );
+      testUntil(
+          domainStatusReasonMatches(domainName, domainNamespace, "ReplicasTooHigh"),
+          getLogger(),
+          "waiting for domain status condition reason ReplicasTooHigh exists"
```

Comment:
Why do the domain still have the ReplicasTooHigh status after you patch the domain replicas to 2?

Reply:
We don't remove the prior status even after rectifying the issue. I guess it is kept as a history